packages/block-editor/src/components/block-variation-picker/index.js

```diff
@@ -43,7 +43,7 @@ function BlockVariationPicker( {
 					<li key={ variation.name }>
 						<Button
 							variant="secondary"
-							icon={ variation.icon }
+							icon={ variation.icon.src || variation.icon }
```

Comment:
Most code guards against icon being undefined using "variation.icon && variation.icon.src ? variation.icon.src : variation.icon". I guess we can follow a similar approach but other that the change looks good and can be merged after the update.